src/lib/component/ImageCarousel.svelte

```diff
@@ -1,48 +1,57 @@
 <script lang="ts">
   import { onMount } from 'svelte'
-  import { pb } from '$lib/pocketbase/pb.js'
+  import { pb, currentUser } from '$lib/pocketbase/pb.js'
   import Skeleton from 'svelte-skeleton/Skeleton.svelte'
+  import { get } from 'svelte/store'
+  import type {
+    ProductResponse,
+    StoreSectionResponse,
+  } from '$lib/pocketbase/pocketbase-types'
+
+  type StoreSectionExpanded = StoreSectionResponse<{
+    products: ProductResponse[]
+  }>
 
-  let productCollectionRecords: any[] = []
   let isLoading = true
-  let index = 0
   let interval: any
+  let index: number
 
-  //we just want primary imge and link url
+  //we just want primary image and link url
   type ProductData = {
     imageUrl: string
     linkUrl: string
   }
 
-  let productArr: ProductData[] = []
-
-  //fetch from pocket host
-  async function fetchProducts() {
-    try {
-      productCollectionRecords = await pb
-        .collection('product')
-        .getFullList(200, {
-          sort: '-created',
-        })
-      productArr = getProductData(productCollectionRecords)
-    } catch (error) {
-      console.error('Error fetching products:', error)
-    } finally {
-      isLoading = false
-      //update state for skelton loader
-    }
-  }
+  let productArr: ProductData[]
+
+  export async function load() {
+    //get current user id
+    const userId = get(currentUser)?.id
+
+    //check if user id exists in allow_preview or in team_groups
+    const filter = userId
+      ? pb.filter('enabled = true || allow_preview ~ {:userId}', { userId })
+      : 'enabled = true'
 
-  //get product data, we want global products that only start with r-ulimate. populate prudctArr with objects, where each contaisn image url and link url
-  function getProductData(productList: any[]): ProductData[] {
-    return productList
-      .filter(
-        (product) => product.enabled && product.slug.startsWith('r-ultimate'), // Filter by slug prefix
-      )
-      .map((product) => ({
+    //get full list of products according to this filter, returns jsonified format
+    const resp = await pb
+      .collection('store_section')
+      .getFullList<StoreSectionExpanded>({
+        filter,
+        expand: 'products',
+      })
+
+    //set our productArr to be an array of objects with imageUrl and linkUrl, as we flatmapped resp
+    productArr = resp.flatMap((section) =>
+      (section.expand?.products ?? []).map((product) => ({
         imageUrl: pb.getFileUrl(product, product.primaryImage),
         linkUrl: '/store/product/' + product.slug,
-      }))
+      })),
+    )
+    //randomize index we start on
+    index = Math.floor(Math.random() * productArr.length)
```

Comment:
2 things:
1. This still doesn't randomize order. I think it would be better to do that
2. There's an "ugly" case where the index starts at the end of the array and immediately loops back around. Try `index=productarr.length - 1` on you local to see what I mean.

You can solve both of these problems by starting the index at 0 and randomizing the array order rather than choosing a random index.